Frends.GoogleDrive.Download/Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.cs

```diff
@@ -1,143 +1,153 @@
+                    {
+                        success = false;
+                        if (options.ThrowErrorOnFailure)
+                            throw new Exception($"Error downloading file: {file.Name}. Error: {result.Exception.Message}");
+                        errorList.Add($"Error downloading file: {file.Name}. Error: {result.Exception.Message}");
+                    }
+
+                    if (success)
+                        resultList.Add(new DownloadResult(file.Id, file.Name));
+                }
+            }
+        }
+        catch (Exception)
+        {
+            throw;
+        }
+        finally
+        {
+            service?.Dispose();
+        }
+
+        return new Result(errorList.Count == 0, resultList, errorList);
+    }
+
+    private static void InputCheck(string serviceAccountKeyJSON, string DestinationDirectory, bool createDir)
+    {
+        if (string.IsNullOrWhiteSpace(serviceAccountKeyJSON))
+            throw new ArgumentException(@"Service account key missing.", nameof(serviceAccountKeyJSON));
+        if (!Directory.Exists(DestinationDirectory))
+        {
+            if (createDir)
+                Directory.CreateDirectory(DestinationDirectory);
+            else
+                throw new ArgumentException(@"Destination directory doesn't exist.", $"{nameof(DestinationDirectory)} Value:{DestinationDirectory}");
+        }
+    }
+
+    private static async Task<DriveService> CreateDriveService(Input input, CancellationToken cancellationToken)
+    {
+        byte[] byteArray;
+        if (File.Exists(input.ServiceAccountKeyJSON))
+            byteArray = await File.ReadAllBytesAsync(input.ServiceAccountKeyJSON, cancellationToken: cancellationToken);
+        else
+            byteArray = Encoding.UTF8.GetBytes(input.ServiceAccountKeyJSON);
+
+        using var stream = new MemoryStream(byteArray);
+        GoogleCredential credential = GoogleCredential.FromStreamAsync(stream, cancellationToken: cancellationToken).Result.CreateScoped(DriveService.ScopeConstants.Drive);
+        var service = new DriveService(new BaseClientService.Initializer
+        {
+            HttpClientInitializer = credential,
+            ApplicationName = "Frends iPaaS"
+        });
+
+        return service;
+    }
+
+    private static IList<Google.Apis.Drive.v3.Data.File> GetFileList(DriveService service, Input input)
+    {
+        FilesResource.ListRequest fileListRequest = service.Files.List();
+        fileListRequest.Fields = "nextPageToken, files(id, kind, name, size, version, createdTime)";
+        fileListRequest.Q = input.FileQuery;
+        fileListRequest.PageSize = 1000;
+
+        // shared drives support
+        fileListRequest.SupportsAllDrives = input.IncludeSharedDrives;
+        fileListRequest.IncludeItemsFromAllDrives = input.IncludeSharedDrives;
+
+        var ret = fileListRequest.Execute();
+        return ret.Files;

```

Comment:
Can be converted to async method

```suggestion
        var ret = await fileListRequest.ExecuteAsync(token);
```